tests/test_authentication.py

```
@@ -146,7 +146,7 @@ def test_get_auth_token_ssh_fail(self, vo):
         except Duplicate:
             pass  # might already exist, can skip
 
-        signature = ssh_sign(PRIVATE_KEY, 'sign_something_else')
+        signature = base64.b64decode(ssh_sign(PRIVATE_KEY, 'sign_something_else'))
 
         result = get_auth_token_ssh(account='root', signature=signature, appid='test', ip='127.0.0.1', vo=vo)
```

I’m not too fond of the situation we ended up in. `ssh_sign()` converts `bytes` to `str`, which you have to convert back to `bytes` to use `get_auth_token_ssh()`.

https://github.com/rucio/rucio/blob/e85183fe6b77841ca01e2d68a1426526bef81b1b/lib/rucio/web/rest/flaskapi/v1/auth.py#L1201-L1229

I wonder if it’s worth propagating the the payload as it is, as an `str`, down to the core layer. There, inside `get_auth_token_ssh()`, we put the decoding operation and raise let it raise a `TypeError` if it fails. The code in the REST layer can then be simplified.

But probably this is going too far, so feel free to ignore.

---

Addressed in this follow-up PR: https://github.com/rucio/rucio/pull/6649